### DISCUSSION TITLE
MaxTokensLimiter usage

### DISCUSSION THREAD

**Discussion by duxnp** [strawberry-graphql/strawberry#3696] [Q&A]
I've been learning GraphQL by reading through the Strawberry documentation. I hit a snag with MaxTokensLimiter. It seems like no matter what I try it doesn't seem to have any effect.

Let's say I take the example from the Getting started page then add the MaxTokensLimiter extension with some extremely low number just to test it out with this very simple example.

```python
schema = strawberry.Schema(
    Query,
    extensions=[
        MaxTokensLimiter(max_token_count=2),
    ],
)
```

If I send the following query from GraphiQL (which I _think_ is 7 tokens), I get the normal response rather than an error message.

```
{
  books {
    title
    author
  }
}
```

Being so new to both Strawberry and GraphQL I assume I'm doing something wrong. I'm wondering if it is unexpected behavior though. I appreciate any help anyone can give.

**Comment by DoctorJohn** [comment]
This might be a bug. I just checked our code and it looks like the limit is not respected when using an async HTTP integration. Are you currently using the `strawberry server` command to run your code?

**Reply by duxnp** [reply]
Yes I'm using the `strawberry server` command to run the Getting Started example.

For my first go at it I was using the FastAPI integration using the `uvicorn` command to run the code.

**Comment by fuJiin** [comment]
Any updates on this? Running into similar issues and it's exposing us to DoS risk.
